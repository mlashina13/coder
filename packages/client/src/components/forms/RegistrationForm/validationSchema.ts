import * as yup from 'yup';

/**
 * Схема валидации
 */
export const validationSchema = yup.object({
  login: yup
    .string()
    .matches(/^[a-zA-Z0-9-_]{3,20}$/g, {
      message:
        'Логин должен быть от 3 до 20 символов, написан латиницей, допускаются цифры, дефис и нижнее подчёркивание',
    })
    .required('Введите логин'),
  password: yup
    .string()
    .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)/g, {
      message:
        'Пароль должен быть от 8 до 40 символов, обязательно хотя бы одна заглавная буква и одна цифра',
    })
    .required('Введите пароль'),
  email: yup
    .string()
    .matches(/^([\w.-])+@([\w.-])+\.([A-Za-z]{2,4})$/, {
      message: 'Почта должна быть написана на латинице, допускаются цифры и спецсимволы',
    })
    .required('Введите e-mail'),
  first_name: yup
    .string()
    .matches(/^[А-ЯA-Z][a-zа-я-]{1,20}$/g, {
      message:
        'Имя должно быть написано на латинице или кириллице, первая буква заглавная, без цифр и спецсимволов',
    })
    .required('Введите имя'),
  second_name: yup
    .string()
    .matches(/^[А-ЯA-Z][a-zа-я-]{1,20}$/g, {
      message:
        'Фамилия должна быть написана на латинице или кириллице, первая буква заглавная, без цифр и спецсимволов',
    })
    .required('Введите фамилию'),
  phone: yup
    .string()
    .matches(/^((8|\+7)[-]?)?(\(?\d{3}\)?[-]?)?[\d-]{7,10}$/, {
      message:
        'Телефон должен быть от 10 до 15 символов, состоять из цифр, может начинается с плюса',
    })
    .required('Введите телефон'),
  passwordRepeat: yup
    .string()
    .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)/g, {
      message:
        'Пароль должен быть от 8 до 40 символов, обязательно хотя бы одна заглавная буква и одна цифра',
    })
    .required('Введите пароль'),
});
