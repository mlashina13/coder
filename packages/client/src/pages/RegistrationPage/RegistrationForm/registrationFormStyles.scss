@import '../../../styles/color';
@import '../../../styles/common';

.regstration-form {
  display: flex;
  flex-direction: column;

  &__input {
    width: 25em;
  }

  button {
    margin-top: $spacing-ml;
  }
}
