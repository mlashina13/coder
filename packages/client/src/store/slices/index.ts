export { errorReducer } from './errorSlice';
export { userReducer } from './userSlice';
export { locationReducer } from './locationSlice';
export { leadboardReducer } from './leaderboardSlice';
export { oauthReducer } from './oauthSlice';
