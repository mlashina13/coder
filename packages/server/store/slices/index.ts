export { errorReducer } from './errorSlice';
// export { userReducer } from './userSlice';
export { locationReducer } from './locationSlice';
