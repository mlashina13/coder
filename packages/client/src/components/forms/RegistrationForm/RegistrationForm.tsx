import { FC, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useFormik } from 'formik';
import { Button, Input } from '../../common';
import { RegistrationData } from '../../../types/common';
import { validationSchema } from './validationSchema';
import { useAppDispatch, useAppSelector } from '../../../hooks';
import { registration } from '../../../services';
import { ROUTER_URLS } from '../../../constants';
import './registrationFormStyles.scss';

/**
 * Модель, описывающая данные формы
 */
interface RegistrationFormData extends RegistrationData {
  passwordRepeat: string;
}

/**
 * Компонент формы регистрации
 */
export const RegistrationForm: FC = () => {
  const { loading, currentUser } = useAppSelector((state) => state.user);
  const dispatch = useAppDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    if (currentUser) {
      navigate(ROUTER_URLS.Main);
    }
  }, [currentUser]);

  const formik = useFormik<RegistrationFormData>({
    initialValues: {
      login: '',
      password: '',
      email: '',
      passwordRepeat: '',
      phone: '',
      second_name: '',
      first_name: '',
    },
    validationSchema,
    onSubmit: ({ password, passwordRepeat, ...rest }) => {
      if (password !== passwordRepeat) {
        formik.setFieldError('passwordRepeat', 'Пароль и повторенный пароль не совпадают!');
      } else {
        dispatch(registration({ ...rest, password }));
      }
    },
  });

  return (
    <form onSubmit={formik.handleSubmit} className='regstration-form'>
      <Input
        name='first_name'
        value={formik.values.first_name}
        onChange={formik.handleChange}
        error={formik.touched.first_name && Boolean(formik.errors.first_name)}
        helperText={formik.touched.first_name && formik.errors.first_name}
        className='regstration-form__input requiredField'
        label='Имя'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='second_name'
        value={formik.values.second_name}
        onChange={formik.handleChange}
        error={formik.touched.second_name && Boolean(formik.errors.second_name)}
        helperText={formik.touched.second_name && formik.errors.second_name}
        className='regstration-form__input requiredField'
        label='Фамилия'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='email'
        value={formik.values.email}
        onChange={formik.handleChange}
        error={formik.touched.email && Boolean(formik.errors.email)}
        helperText={formik.touched.email && formik.errors.email}
        className='regstration-form__input requiredField'
        label='E-mail'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='phone'
        value={formik.values.phone}
        onChange={formik.handleChange}
        error={formik.touched.phone && Boolean(formik.errors.phone)}
        helperText={formik.touched.phone && formik.errors.phone}
        className='regstration-form__input requiredField'
        label='Телефон'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='login'
        value={formik.values.login}
        onChange={formik.handleChange}
        error={formik.touched.login && Boolean(formik.errors.login)}
        helperText={formik.touched.login && formik.errors.login}
        className='regstration-form__input requiredField'
        label='Логин'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='password'
        value={formik.values.password}
        onChange={formik.handleChange}
        onBlur={formik.handleBlur}
        error={formik.touched.password && Boolean(formik.errors.password)}
        helperText={formik.touched.password && formik.errors.password}
        type='password'
        className='regstration-form__input requiredField'
        label='Пароль'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='passwordRepeat'
        value={formik.values.passwordRepeat}
        onChange={formik.handleChange}
        onBlur={formik.handleBlur}
        error={formik.touched.passwordRepeat && Boolean(formik.errors.passwordRepeat)}
        helperText={formik.touched.passwordRepeat && formik.errors.passwordRepeat}
        type='password'
        className='regstration-form__input requiredField'
        label='Подтвердите пароль'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Button label='Регистрация' type='submit' disabled={loading} />
    </form>
  );
};
