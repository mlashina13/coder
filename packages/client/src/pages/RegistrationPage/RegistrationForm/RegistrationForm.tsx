import { useFormik } from 'formik';
import * as yup from 'yup';

import { Button, Input } from '../../../components';
import { Auth } from '../../../services';
import { useUserStore } from '../../../stores';

import './registrationFormStyles.scss';

/** схема валидации */
const validationSchema = yup.object({
  login: yup
    .string()
    /** проверка через регулярное выражение логина */
    .matches(/^[a-zA-Z0-9-_]{3,20}$/g, {
      message:
        'Логин должен быть от 3 до 20 символов, написан латиницей, допускаются цифры, дефис и нижнее подчёркивание',
    })
    /** указываем на обязательность логина */
    .required('Введите логин'),
  password: yup
    .string()
    /** проверка через регулярное выражение пароля */
    .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)/g, {
      message:
        'Пароль должен быть от 8 до 40 символов, обязательно хотя бы одна заглавная буква и одна цифра',
    })
    /** указываем на обязательность пароля */
    .required('Введите пароль'),
  email: yup
    .string()
    /** проверка через регулярное выражение электронной почты */
    .matches(/^([\w.-])+@([\w.-])+\.([A-Za-z]{2,4})$/, {
      message: 'Почта должна быть написана на латинице, допускаются цифры и спецсимволы',
    })
    /** указываем на обязательность электронной почты */
    .required('Введите e-mail'),
  first_name: yup
    .string()
    /** проверка через регулярное выражение имени */
    .matches(/^[А-ЯA-Z][a-zа-я-]{1,20}$/g, {
      message:
        'Имя должно быть написано на латинице или кириллице, первая буква заглавная, без цифр и спецсимволов',
    })
    /** указываем на обязательность имени */
    .required('Введите имя'),
  second_name: yup
    .string()
    /** проверка через регулярное выражение фамилии */
    .matches(/^[А-ЯA-Z][a-zа-я-]{1,20}$/g, {
      message:
        'Фамилия должна быть написана на латинице или кириллице, первая буква заглавная, без цифр и спецсимволов',
    })
    /** указываем на обязательность фамилии */
    .required('Введите фамилию'),
  phone: yup
    .string()
    /** проверка через регулярное выражение телефона */
    .matches(/^((8|\+7)[-]?)?(\(?\d{3}\)?[-]?)?[\d-]{7,10}$/, {
      message:
        'Телефон должен быть от 10 до 15 символов, состоять из цифр, может начинается с плюса',
    })
    /** указываем на обязательность телефона */
    .required('Введите телефон'),
  passwordRepeat: yup
    .string()
    /** проверка через регулярное выражение пароля */
    .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)/g, {
      message:
        'Пароль должен быть от 8 до 40 символов, обязательно хотя бы одна заглавная буква и одна цифра',
    })
    /** указываем на обязательность пароля */
    .required('Введите пароль'),
});

export const RegistrationForm = () => {
  const { setUserData } = useUserStore();
  const formik = useFormik({
    initialValues: {
      login: '',
      password: '',
      email: '',
      passwordRepeat: '',
      phone: '',
      second_name: '',
      first_name: '',
    },
    validationSchema,
    onSubmit: ({ password, passwordRepeat, ...rest }) => {
      if (password !== passwordRepeat) {
        console.error('Введите корректный пароль');
      } else {
        Auth.registration({ ...rest, password }).then(() => {
          Auth.getAuthUser().then((data) => setUserData?.(data));
        });
      }
    },
  });

  return (
    <form onSubmit={formik.handleSubmit} className='regstration-form'>
      <Input
        name='first_name'
        value={formik.values.first_name}
        onChange={formik.handleChange}
        error={formik.touched.first_name && Boolean(formik.errors.first_name)}
        helperText={formik.touched.first_name && formik.errors.first_name}
        className='regstration-form__input requiredField'
        label='Имя'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='second_name'
        value={formik.values.second_name}
        onChange={formik.handleChange}
        error={formik.touched.second_name && Boolean(formik.errors.second_name)}
        helperText={formik.touched.second_name && formik.errors.second_name}
        className='regstration-form__input requiredField'
        label='Фамилия'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='email'
        value={formik.values.email}
        onChange={formik.handleChange}
        error={formik.touched.email && Boolean(formik.errors.email)}
        helperText={formik.touched.email && formik.errors.email}
        className='regstration-form__input requiredField'
        label='E-mail'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='phone'
        value={formik.values.phone}
        onChange={formik.handleChange}
        error={formik.touched.phone && Boolean(formik.errors.phone)}
        helperText={formik.touched.phone && formik.errors.phone}
        className='regstration-form__input requiredField'
        label='Телефон'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='login'
        value={formik.values.login}
        onChange={formik.handleChange}
        error={formik.touched.login && Boolean(formik.errors.login)}
        helperText={formik.touched.login && formik.errors.login}
        className='regstration-form__input requiredField'
        label='Логин'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='password'
        value={formik.values.password}
        onChange={formik.handleChange}
        onBlur={formik.handleBlur}
        error={formik.touched.password && Boolean(formik.errors.password)}
        helperText={formik.touched.password && formik.errors.password}
        type='password'
        className='regstration-form__input requiredField'
        label='Пароль'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Input
        name='passwordRepeat'
        value={formik.values.passwordRepeat}
        onChange={formik.handleChange}
        onBlur={formik.handleBlur}
        error={formik.touched.passwordRepeat && Boolean(formik.errors.passwordRepeat)}
        helperText={formik.touched.passwordRepeat && formik.errors.passwordRepeat}
        type='password'
        className='regstration-form__input requiredField'
        label='Подтвердите пароль'
        autoComplete='off'
        size='small'
        variant='filled'
      />
      <Button label='Регистрация' type='submit' />
    </form>
  );
};
