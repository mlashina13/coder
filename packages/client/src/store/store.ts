import { configureStore } from '@reduxjs/toolkit';
import { errorReducer, userReducer, locationReducer } from './slices';

declare global {
  interface Window {
    // В d.ts нам неважно, что это за тип,
    // так как он сразу попадает в redux store на клиенте
    __PRELOADED_STATE__?: object;
  }
}

// console.log(serverState);

/**
 * Хранилище приложения
 */
export const store = configureStore({
  reducer: {
    error: errorReducer,
    user: userReducer,
    location: locationReducer,
  },
});

export const createStore = () => {
  const serverState = window.__PRELOADED_STATE__ || {};

  return configureStore({
    reducer: {
      error: errorReducer,
      user: userReducer,
      location: locationReducer,
    },
    preloadedState: serverState,
  });
};

/**
 * Тип стейта приложения
 */
export type RootState = ReturnType<typeof store.getState>;

/**
 * Тип диспетчера
 */
export type AppDispatch = typeof store.dispatch;
