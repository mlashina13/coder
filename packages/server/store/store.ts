// eslint-disable-next-line import/no-extraneous-dependencies
import {
  EnhancedStore,
  StoreEnhancer,
  ThunkDispatch,
  Tuple,
  UnknownAction,
  configureStore,
} from '@reduxjs/toolkit';
import { errorReducer, locationReducer } from './slices';

export type StoreType = EnhancedStore<
  Record<string, unknown>,
  UnknownAction,
  Tuple<
    [
      StoreEnhancer<{
        dispatch: ThunkDispatch<Record<string, unknown>, undefined, UnknownAction>;
      }>,
      StoreEnhancer
    ]
  >
>;

/**
 * Хранилище приложения
 */
export const store: StoreType = configureStore({
  reducer: {
    error: errorReducer,
    // user: userReducer,
    location: locationReducer,
  },
});

/**
 * Тип стейта приложения
 */
export type RootState = ReturnType<typeof store.getState>;

/**
 * Тип диспетчера
 */
export type AppDispatch = typeof store.dispatch;
