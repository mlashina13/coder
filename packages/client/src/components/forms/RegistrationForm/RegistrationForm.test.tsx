/* eslint-disable import/no-extraneous-dependencies */
/**
 * @jest-environment node
 */
import React from 'react';
import { mount, render, configure } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import { store } from '../../../store';
import { RegistrationForm } from './RegistrationForm';

configure({ adapter: new Adapter() });
describe('Registration Form', () => {
  test('Должен верно отрисовываться', () => {
    const WrappedForm = () => (
      <Provider store={store}>
        <BrowserRouter>
          <RegistrationForm />
        </BrowserRouter>
      </Provider>
    );

    const tree = render(<WrappedForm />);
    expect(tree).toMatchSnapshot();
  });

  describe('Проверка ф-ти', () => {
    const WrappedForm = () => (
      <Provider store={store}>
        <BrowserRouter>
          <RegistrationForm />
        </BrowserRouter>
      </Provider>
    );

    test('Должен обновлять поле с паролем', () => {
      const passwordInput = mount<React.FC>(<WrappedForm />)
        .find(RegistrationForm)
        .find('input[name="password"]');

      act(() => {
        passwordInput.simulate('change', {
          persist: () => undefined,
          target: {
            name: 'password',
            value: '12345678',
          },
        });
      });
      expect(passwordInput.html()).toMatch('12345678');
    });

    test('Должен обновлять поле с логином', () => {
      const loginInput = mount<React.FC>(<WrappedForm />)
        .find(RegistrationForm)
        .find("input[name='login']");

      act(() => {
        loginInput.simulate('change', {
          persist: () => undefined,
          target: {
            name: 'login',
            value: 'newName',
          },
        });
      });

      expect(loginInput.html()).toMatch('newName');
    });

    test('Должен обновлять поле с почтой', () => {
      const passwordInput = mount<React.FC>(<WrappedForm />)
        .find(RegistrationForm)
        .find('input[name="email"]');

      act(() => {
        passwordInput.simulate('change', {
          persist: () => undefined,
          target: {
            name: 'email',
            value: 'email@email.ru',
          },
        });
      });
      expect(passwordInput.html()).toMatch('email@email.ru');
    });

    test('Должен обновлять поле "повторить пароль"', () => {
      const passwordInput = mount<React.FC>(<WrappedForm />)
        .find(RegistrationForm)
        .find('input[name="passwordRepeat"]');

      act(() => {
        passwordInput.simulate('change', {
          persist: () => undefined,
          target: {
            name: 'passwordRepeat',
            value: '87654321',
          },
        });
      });
      expect(passwordInput.html()).toMatch('87654321');
    });

    test('Должен обновлять поле с телефоном', () => {
      const passwordInput = mount<React.FC>(<WrappedForm />)
        .find(RegistrationForm)
        .find('input[name="phone"]');

      act(() => {
        passwordInput.simulate('change', {
          persist: () => undefined,
          target: {
            name: 'phone',
            value: '8888888888',
          },
        });
      });
      expect(passwordInput.html()).toMatch('8888888888');
    });

    test('Должен обновлять поле с фамилией', () => {
      const passwordInput = mount<React.FC>(<WrappedForm />)
        .find(RegistrationForm)
        .find('input[name="second_name"]');

      act(() => {
        passwordInput.simulate('change', {
          persist: () => undefined,
          target: {
            name: 'second_name',
            value: 'Ivanovich',
          },
        });
      });
      expect(passwordInput.html()).toMatch('Ivanovich');
    });

    test('Должен обновлять поле с именем', () => {
      const passwordInput = mount<React.FC>(<WrappedForm />)
        .find(RegistrationForm)
        .find('input[name="first_name"]');

      act(() => {
        passwordInput.simulate('change', {
          persist: () => undefined,
          target: {
            name: 'first_name',
            value: 'Ivan',
          },
        });
      });
      expect(passwordInput.html()).toMatch('Ivan');
    });
  });
});
