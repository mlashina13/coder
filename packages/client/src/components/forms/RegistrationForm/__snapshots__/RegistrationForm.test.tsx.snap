// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Должен верно отрисовываться 1`] = `
LoadedCheerio {
  "0": <form
    class="regstration-form"
  >
    <div
      class="MuiFormControl-root MuiTextField-root input regstration-form__input requiredField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":R1:"
        id=":R1:-label"
      >
        Имя
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1ff8729-MuiInputBase-root-MuiFilledInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
          id=":R1:"
          name="first_name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root input regstration-form__input requiredField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":R2:"
        id=":R2:-label"
      >
        Фамилия
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1ff8729-MuiInputBase-root-MuiFilledInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
          id=":R2:"
          name="second_name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root input regstration-form__input requiredField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":R3:"
        id=":R3:-label"
      >
        E-mail
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1ff8729-MuiInputBase-root-MuiFilledInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
          id=":R3:"
          name="email"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root input regstration-form__input requiredField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":R4:"
        id=":R4:-label"
      >
        Телефон
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1ff8729-MuiInputBase-root-MuiFilledInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
          id=":R4:"
          name="phone"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root input regstration-form__input requiredField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":R5:"
        id=":R5:-label"
      >
        Логин
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1ff8729-MuiInputBase-root-MuiFilledInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
          id=":R5:"
          name="login"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root input regstration-form__input requiredField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":R6:"
        id=":R6:-label"
      >
        Пароль
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1ff8729-MuiInputBase-root-MuiFilledInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
          id=":R6:"
          name="password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root input regstration-form__input requiredField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":R7:"
        id=":R7:-label"
      >
        Подтвердите пароль
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1ff8729-MuiInputBase-root-MuiFilledInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
          id=":R7:"
          name="passwordRepeat"
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium button css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Регистрация
    </button>
  </form>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
